src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -245,6 +270,11 @@ public function merge(AttributeMetadataInterface $attributeMetadata)
             $this->serializedName = $attributeMetadata->getSerializedName();
         }
 
+        // Overwrite only if not defined
+        if (null === $this->serializedPath) {
+            $this->serializedPath = $attributeMetadata->getSerializedPath();
```

Comment:
`$this->serializedPath ??= `
(let's upgrade the other ones also, but please rebase your PR on 6.1 before doing so because we already moved to `??=` I think)